internal/gomodules/gomodules.go

```diff
@@ -23,7 +23,7 @@ func FindDependencies(wd string) (depbot.Dependencies, error) {
 	pths := []string{}
 
 	filepath.WalkDir(wd, func(path string, d fs.DirEntry, err error) error {
-		if strings.Contains(path, goDependencyFile) {
+		if strings.HasSuffix(path, "/"+goDependencyFile) {
```

Comment:
@bcuadrad1 this method should use the [filepath.Base](https://pkg.go.dev/path/filepath#Base) function.